### ISSUE TITLE
Projects stopped being detected after "nx migrate latest" (any version after 16.6.0)

### ISSUE THREAD

**Issue by josetapadas** [nrwl/nx#18821]
### Current Behavior

So basically after trying to migrate nx to the latest version via:

nx migrate latest
And running the migrations and yarn my projects in the monorepo stopped being detected by int:

nx show projects
Simply outputs blank.

The setup is:

nx v16.7.4 (local)
yarn 3.6.1
node 16.19.1
And I use asdf on an M1 mac.

If I revert back to a previous version of nx (15.9.2) all work fine. I'm busting yarn and nx caches as well as deleting yarn.lock and/or removing node_modules without luck.

Any idea/suggestion on how we can trace this or how to fix it?

### Expected Behavior

nx show projects showing all the apps and libs I have exposed.

### Steps to Reproduce

1. nx show projects (list all the projects)
2. nx reset
3. nx migrate latest
4. yarn
5. nx show projects (empty list!)

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 16.19.1
   OS     : darwin-arm64
   yarn   : 3.6.3

   nx                 : 16.7.4
   lerna              : 6.6.2
   @nx/js             : 16.7.4
   @nx/jest           : 16.7.4
   @nx/linter         : 16.7.4
   @nx/workspace      : 16.7.4
   @nx/cypress        : 16.7.4
   @nx/devkit         : 16.7.4
   @nrwl/devkit       : 15.9.5
   @nx/eslint-plugin  : 16.7.4
   @nx/express        : 16.7.4
   @nx/next           : 16.7.4
   @nx/node           : 16.7.4
   @nx/react          : 16.7.4
   @nx/storybook      : 16.7.4
   @nrwl/tao          : 16.7.4
   @nx/vite           : 16.7.4
   @nx/web            : 16.7.4
   @nx/webpack        : 16.7.4
   typescript         : 5.1.6
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/devkit@15.9.5

   To fix this, run `nx migrate nx@16.7.4`
```

### Package Manager Version

3.6.3

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

This was working with nx 15.9.2 and works with 16.5.0

**Comment by FrozenPandaz** [comment]
Can you please provide a repo where you see the issue? What else can you tell me about your workspace? What version are you migrating from?

**Comment by josetapadas** [comment]
After a bit of digging on my env I could find the root of it.

TL;DR please look into your ignore files *both on your project directory but as well your home directory* and check if they contain either:

- *.json
- project.json
- package.json

If so then nx won't be finding your apps and libs as their `project.json` files will be ignored.

Basically on my specific case I had a `.ignore` file, on macOS, that I really don't know what tool created it (intelliJ, solargraph, ?) hence my files started being ignored after 16.6.0.

After removing those ignore entries it all started working properly.
